std/math/exponential.d

```diff
@@ -16,8 +16,827 @@ License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
 Authors:   $(HTTP digitalmars.com, Walter Bright), Don Clugston,
            Conversion of CEPHES math library to D by Iain Buclaw and David Nadlinger
 Source: $(PHOBOSSRC std/math/exponential.d)
+
+Macros:
+    TABLE_SV = <table border="1" cellpadding="4" cellspacing="0">
+               <caption>Special Values</caption>
+               $0</table>
+    NAN = $(RED NAN)
+    PLUSMN = &plusmn;
+    INFIN = &infin;
+    PLUSMNINF = &plusmn;&infin;
+    LT = &lt;
+    GT = &gt;
  */
 
 module std.math.exponential;
 
-// Will contain functions like pow, exp, log2
+import std.traits :  isFloatingPoint, isIntegral, Largest, Unqual;
+
+/**
+ * Compute the value of x $(SUPERSCRIPT n), where n is an integer
+ */
+Unqual!F pow(F, G)(F x, G n) @nogc @trusted pure nothrow
+if (isFloatingPoint!(F) && isIntegral!(G))
+{
+    import std.math : abs, floor, isNaN, log2;
+    import std.traits : Unsigned;
+
+    // NaN ^^ 0 is an exception defined by IEEE (yields 1 instead of NaN)
+    if (isNaN(x)) return n ? x : 1.0;
+
+    import std.traits : Unsigned;
```

Comment:
You're importing this twice here.
```suggestion
    import std.traits : Unsigned;

    // NaN ^^ 0 is an exception defined by IEEE (yields 1 instead of NaN)
    if (isNaN(x)) return n ? x : 1.0;

```